src/audio_core/hle/wmf_decoder.cpp

```diff
@@ -0,0 +1,254 @@
+        return {};
+    }
+    u8* data = memory.GetFCRAMPointer(request.src_addr - Memory::FCRAM_PADDR);
+
+    std::array<std::vector<u8>, 2> out_streams;
+    IMFSample* sample = NULL;
+    ADTSData adts_header;
+    char* aac_tag = (char*)calloc(1, 14);
+    int input_status = 0;
+
+    if (detect_mediatype((char*)data, request.size, &adts_header, &aac_tag) != 0) {
+        LOG_ERROR(Audio_DSP, "Unable to deduce decoding parameters from ADTS stream");
+        return response;
+    }
+
+    if (!selected) {
+        LOG_DEBUG(Audio_DSP, "New ADTS stream: channels = {}, sample rate = {}",
+                  adts_header.channels, adts_header.samplerate);
+        select_input_mediatype(transform, in_stream_id, adts_header, (UINT8*)aac_tag, 14);
+        select_output_mediatype(transform, out_stream_id);
+        send_sample(transform, in_stream_id, NULL);
+        // cache the result from detect_mediatype and call select_*_mediatype only once
+        // This could increase performance very slightly
+        transform->ProcessMessage(MFT_MESSAGE_NOTIFY_BEGIN_STREAMING, 0);
+        selected = true;
+    }
+
+    sample = create_sample((void*)data, request.size, 1, 0);
+    sample->SetUINT32(MFSampleExtension_CleanPoint, 1);
+
+    while (true) {
+        input_status = send_sample(transform, in_stream_id, sample);
+
+        if (DecodingLoop(adts_header, out_streams) < 0) {
+            // if the decode issues is caused by MFT not accepting new samples, try again
+            // NOTICE: you are required to check the output even if you already knew/guessed
+            // MFT didn't accept the input sample
+            if (input_status == 1) {
+                // try again
+                continue;
+            }
+
+            return response;
+        }
+
+        break; // jump out of the loop if at least we don't have obvious issues
+    }
+
+    if (out_streams[0].size() != 0) {
+        if (request.dst_addr_ch0 < Memory::FCRAM_PADDR ||
+            request.dst_addr_ch0 + out_streams[0].size() >
+                Memory::FCRAM_PADDR + Memory::FCRAM_SIZE) {
+            LOG_ERROR(Audio_DSP, "Got out of bounds dst_addr_ch0 {:08x}", request.dst_addr_ch0);
+            return {};
+        }
+        std::memcpy(memory.GetFCRAMPointer(request.dst_addr_ch0 - Memory::FCRAM_PADDR),
+                    out_streams[0].data(), out_streams[0].size());
+    }
+
+    if (out_streams[1].size() != 0) {
+        if (request.dst_addr_ch1 < Memory::FCRAM_PADDR ||
+            request.dst_addr_ch1 + out_streams[1].size() >
+                Memory::FCRAM_PADDR + Memory::FCRAM_SIZE) {
+            LOG_ERROR(Audio_DSP, "Got out of bounds dst_addr_ch1 {:08x}", request.dst_addr_ch1);
+            return {};
+        }
+        std::memcpy(memory.GetFCRAMPointer(request.dst_addr_ch1 - Memory::FCRAM_PADDR),
+                    out_streams[1].data(), out_streams[1].size());
+    }
+
+    response.num_channels = adts_header.channels;
```

Comment:
Probably move this and the adts_header parsing before the init check

Reply:
I'll address this